MTA_limit_trends/Comp/run_comp_grad_data_update.py

```diff
@@ -6,59 +6,57 @@
 #                                                                                           #
 #           author: t. isobe (tisobe@cfa.harvard.edu)                                       #
 #                                                                                           #
-#           last update: Feb 02, 2021                                                       #
+#           last update: May 06, 2024                                                       #
 #                                                                                           #
 #############################################################################################
 
 import os
 import sys
 import re
-import string
 import time
 import numpy
 import astropy.io.fits  as pyfits
 from astropy.io.fits import Column
 import Ska.engarchive.fetch as fetch
 import Chandra.Time
+import traceback
+import signal
+import argparse
 import getpass
 #
-#--- reading directory list
+#--- Define Directory Pathing
 #
-path = '/data/mta/Script/MTA_limit_trends/Scripts/house_keeping/dir_list'
-with open(path, 'r') as f:
-    data = [line.strip() for line in f.readlines()]
+BIN_DIR = "/data/mta/Script/MTA_limit_trends/Scripts"
+MTA_DIR = "/data/mta4/Script/Python3.11/MTA"
+DEPOSIT_DIR = "/data/mta/Script/MTA_limit_trends/Deposit"
+COMP_DIR = f"{DEPOSIT_DIR}/Comp_save"
+GRAD_DIR = f"{DEPOSIT_DIR}/Grad_save"
+DATA_DIR = "/data/mta/Script/MTA_limit_trends/Data"
 
-for ent in data:
-    atemp = re.split(':', ent)
-    var  = atemp[1].strip()
-    line = atemp[0].strip()
-    exec("%s = %s" %(var, line))
 #
 #--- append path to a private folder
 #
-sys.path.append(bin_dir)
-sys.path.append(mta_dir)
+sys.path.append(BIN_DIR)
+sys.path.append(MTA_DIR)
 #
 #--- import several functions
 #
-import mta_common_functions     as mcf  #---- contains other functions commonly used in MTA scripts
-import envelope_common_function as ecf  #---- contains other functions commonly used in envelope
 import fits_operation           as mfo  #---- fits operation collection
 import read_limit_table         as rlt  #---- read limit table and create msid<--> limit dict
+
 #
-#--- other path setting
+#--- Define globals
 #
-comp_dir  = deposit_dir + 'Comp_save/'
-grad_dir  = deposit_dir + 'Grad_save/'
+FULL_GEN = False #Used for determining whether to generate the fill fits files or build off the last time entry
+DTYPE = ['long', 'short', 'week']
```

Comment:
These global variables are used purely for testing convenience. The DTYPE option defining the subsections of time with which to generate the smaller data files from the full data files located in `/data/mta/Script/MTA_limit_trends/Deposit/Comp_save`

FULL_GEN as a boolean determines whether the script should delete the smaller section fits files and regenerate them from the beginning, rather than appending the new data to the old file.